src/main/java/com/appium/executor/MyTestExecutor.java

```diff
@@ -104,8 +104,20 @@ public boolean runMethodParallelAppium(List<String> test, String pack, int devic
         int a = 0;
         Collection<URL> urls = ClasspathHelper.forPackage(items.get(a));
         Iterator<URL> iter = urls.iterator();
-        URL url = iter.next();
-        urls.clear();
+
+        URL url = null;
+
+        while (iter.hasNext()) {
```

Comment:
what are these changes for?

Reply:
URL newUrl;
        List<URL> newUrls = new ArrayList<>();
        Collections.addAll(items, pack.split("\\s*,\\s*"));
        int a = 0;
        Collection<URL> urls = ClasspathHelper.forPackage(items.get(a));

If you have maintained same package names in two places say
src/main/java/a/b/c
src/test/java/a/b/c
Then, ClasspathHelper.forPackage(items.get(a)); will return you two urls, one pointing to the main path and the other one to test path.

And earlier we used to take the first url by default and then fetch the methods having @Test annotation using reflections.

As the first url is of Main path, testng would alwatys returned empty test methods.

So have added a check to pick up the Test path url to fetch the test methods.

This would solve the issue #308.

Reply:
src/main/java
src/test/java 

mvn as build tool will pick the tests only from  /test/ folder. Why are we handling this then ?

Reply:
Mvn will only execute tests provided in testng.xml.

When we are constructing testng.xml, we provide the package name as parameter to reflection class ClasspathHelper

 Collection<URL> urls = ClasspathHelper.forPackage(items.get(a));

forpackage(packagename) will return you all distinct collection of urls for given package name

If there are similar package name in main and test (say src/main/java/a/b/c and src/main/test/a/b/c)

then it return two urls, one is from main and one is from test folder

And we hard coded to pick the first url in the list (which points to main folder)

This would result in empty tests in testng.xml 

Hence add a check to pick the test folder. To replicate it please refer the solution provided in #308